src/Pool.sol

```diff
@@ -351,16 +348,16 @@ abstract contract Pool {
     {
         if (count == 0) return 0;
         Receiver storage receiver = receivers[receiverAddr];
-        int128 fundsPerCycle = int128(receiver.lastFundsPerCycle);
         uint64 cycle = receiver.nextCollectedCycle;
+        int128 cycleAmt = 0;
         for (uint256 i = 0; i < count; i++) {
-            fundsPerCycle += receiver.amtDeltas[cycle].thisCycle;
-            collectedAmt += uint128(fundsPerCycle);
-            fundsPerCycle += receiver.amtDeltas[cycle].nextCycle;
+            cycleAmt += receiver.amtDeltas[cycle].thisCycle;
+            collectedAmt += uint128(cycleAmt);
+            cycleAmt += receiver.amtDeltas[cycle].nextCycle;
             delete receiver.amtDeltas[cycle];
             cycle++;
         }
-        receiver.lastFundsPerCycle = uint128(fundsPerCycle);
+        if (cycleAmt != 0) receiver.amtDeltas[cycle].thisCycle += cycleAmt;
```

Comment:
💯

Reply:
Maybe we should add a comment somewhere. 
If it is the next cycle to collect the delta mapping doesn't store the delta instead it includes the cycleAmt

Reply:
I've just tried it in `_flushCyclesInternal` and the whole tests started using 1048 gas more. I don't understand why. It'd be a good price for cleaner code, but IMO it's basically the same.